### ISSUE TITLE
Monstertracker message out of boundaries

### ISSUE THREAD

**Issue by aeskathon** [Excrulon/Tree-of-Savior-Lua-Mods#85]
Sometimes when monster name is too long the message ending goes behind the screen edge.
Is it possible to change offset of the whole frame based on message length?

A bit unrelated: I am willing to participate in evolution of this addons, but I don't know what methods and functions can be used. While I can study lua syntax myself, I still need information on the framework. Is there any database or smth that can help me on this matter?

**Comment by Excrulon** [comment]
Yeah. I'm making it show up in top middle instead and also going to have a
window that lists all monsters on current map and their kill count if
incomplete.

**Comment by fiote** [comment]
About getting started, you may want to take a look at my [cwAPI](https://github.com/fiote/treeofsavior-addons/blob/master/cwapi/cwapi.lua) (still need to create a wiki for that, though).

**Comment by aeskathon** [comment]
Excrulon, thank you for the answer.

fiote, it is really nice, that you are creating an API for this whole mess :) I'll look into it.
Still, I was more interested in a lower layer. I'll post an example
```
local topFrame = ui.GetFrame('skilltree');
topFrame:SetUserValue("CLICK_ABIL_ACTIVE_TIME",imcTime.GetAppTime()-10);
```
So, what kind of object is ui? What methods does it have? What kind of object is returned by GetFrame (well, it obviously is a frame, but you can deduce almost nothing from it). What do the parameters in SetUserValue mean?
So, the main question is 'where did you get all this info that enables you to write this code?'

**Comment by Excrulon** [comment]
by dumping _G and any other table to learn what's in it and also reading
imc's code to learn how they use things.

**Comment by aeskathon** [comment]
So it's basically reverse engineering, right?

**Comment by Excrulon** [comment]
Yep. Will add more to my wiki page as I go to try to help people.

**Comment by aeskathon** [comment]
Splendid! It would surely be beneficial to share the knowledge and thus bring more people to add-on development community.

Of course it would be much better if IMC published the documention on their API, but I guess it won't happen ever.
